builtin/logical/ssh/path_sign.go

```diff
@@ -285,11 +285,21 @@ func (b *backend) calculateKeyId(data *framework.FieldData, req *logical.Request
 		keyIDFormat = role.KeyIDFormat
 	}
 
-	keyID := substQuery(keyIDFormat, map[string]string{
+	substMap := map[string]string{
 		"token_display_name": req.DisplayName,
 		"role_name":          data.Get("role").(string),
 		"public_key_hash":    fmt.Sprintf("%x", sha256.Sum256(pubKey.Marshal())),
-	})
+	}
+
+	entity, _ := b.System().EntityInfo(req.EntityID)
+
+	if entity != nil {
+		for key, value := range entity.Metadata {
+			substMap[fmt.Sprintf("token_meta_%s", key)] = value
```

Comment:
...this is entity metadata, not token metadata.

You'd want to put the metadata into the Alias info that is returned from AWS, then access that Alias here. Except a challenge is that you don't really know what alias it is, so you can't do this just from code; you'd need some kind of parameter that can be set on the role to indicate what alias it should source key info from, and which keys from that alias' metadata.

Reply:
Thanks, I'll look into Alias, looks more challenging than what I expected, but I'll try

Reply:
@jefferai So I have something working now, with very simple alias selection rule: it takes the first one if any. Would it be enough to merge it as is? I'm not sure whether having several entity aliases is a typical use case?
I could add more criteria to filter the aliases, but I don't think I can be sure only one alias will match. If we just wand to add a filter, I think it would make more sense to have a filter on either the alias mount_path, the alias mount_type or maybe on a pattern of the alias name (in any case I would have to take the first one matching the condition)
What do you think?

Reply:
@jefferai congrats on the release of vault 1.0 :) kind bump for my questions above :)